src/EditorFeatures/Core.Wpf/InlineRename/UI/SmartRename/SmartRenameViewModel.cs

```diff
@@ -102,12 +106,17 @@ private void SessionPropertyChanged(object sender, PropertyChangedEventArgs e)
         // _smartRenameSession.SuggestedNames is a normal list. We need to convert it to ObservableCollection to bind to UI Element.
         if (e.PropertyName == nameof(_smartRenameSession.SuggestedNames))
         {
+            var textInputBackup = BaseViewModel.IdentifierText;
+
             SuggestedNames.Clear();
             foreach (var name in _smartRenameSession.SuggestedNames)
             {
                 SuggestedNames.Add(name);
             }
 
+            // Changing the list may have changed the text in the text box. We need to restore it.
```

Comment:
Why this backup is needed?

Reply:
As explained on the comment, changing the SuggestedNames list makes the combobox internally updating its Text property, which may clear IdentifierName. So we do a backup to make sure it stays the same as before updating the list.

Reply:
Emmm, this sounds weird, is it caused because of the ItemSource getting changed? Is it possible to override this behavior? I just feel the backup seems like a hack

Reply:
It's because of the binding on Text property, which internally to WPF reflect the selected item from ItemSource. I did search how to workaround it in a cleaner manner. I thought of adding a property in Smart Rename View Model that would somehow get `IdentifierName` to ignore the change, but I'm not fan of the complexity it added. 

Note that this issue only repro when:
1. User open Rename
2. Click on the Get Suggestion button
3. Select an item
4. Then click on the Get Suggestion button again.

Reply:
OK, I get the ideas here